src/smoosh/src/smoosh_channel.erl

```diff
@@ -194,18 +231,144 @@ handle_info(check_window, State0) ->
         end,
     erlang:send_after(60 * 1000, self(), check_window),
     {noreply, FinalState};
-handle_info(pause, State0) ->
-    {ok, State} = code_change(nil, State0, nil),
+handle_info(start_recovery, #state{name = Name, waiting = Waiting0} = State0) ->
+    RecActive = recover(active_file_name(Name)),
+    Waiting1 = lists:foldl(
+        fun(DbName, Acc) ->
+            case couch_db:is_compacting(DbName) of
+                true ->
+                    Priority = smoosh_server:get_priority(Name, DbName),
+                    smoosh_priority_queue:in(DbName, Priority, Priority, Acc);
+                false ->
+                    Acc
+            end
+        end,
+        Waiting0,
+        RecActive
+    ),
+    State1 = maybe_start_compaction(State0#state{paused = false, waiting = Waiting1}),
+    couch_log:notice(
+        "~p Previously active compaction jobs (if any) have been successfully recovered and restarted.",
+        [?MODULE]
+    ),
+    erlang:send_after(?ACTIVATE_DELAY_IN_MSEC, self(), activate),
```

Comment:
We might want to speed up the activation event by checking the state of `waiting`. 

```
maybe_start_compaction(State) ->
     ....
        length(State#state.active) + length(State#state.starting) < Concurrency ->
            case smoosh_priority_queue:out(State#state.waiting) of
                false ->
                    maybe_activate(State);
```

```
maybe_activate(state#{activated = true} = State) ->
     State;
maybe_activate(State) ->
     activate(State).

activate(State) ->
     %% content of the `handle_info(activate, ...` clause
```

```
handle_info(activate, #state{} = State) ->
     activate(State);
```

Reply:
Doing it this way will cause an infinite loop:
1. Enter `maybe_start_compaction` in `activate`.
2. State is not activated so it calls `activate` again, and this continues.